File: src/main/java/tn/cashfly/entities/ENTREPRISE.java
package tn.cashfly.entities;

import java.time.LocalDate;

public class ENTREPRISE {

    private int idEntreprise;
    private String nom;
    private String secteur;
    private String formeJuridique;
    private LocalDate dateCreation;
    private double capital;
    private int idProprietaire; // FK to Utilisateur

    // Constructor for new Entreprise
    public ENTREPRISE(String nom, String secteur, String formeJuridique,
                      LocalDate dateCreation, double capital, int idProprietaire) {
        this.nom = nom;
        this.secteur = secteur;
        this.formeJuridique = formeJuridique;
        this.dateCreation = dateCreation;
        this.capital = capital;
        this.idProprietaire = idProprietaire;
    }

    // Constructor for Entreprise loaded from DB
    public ENTREPRISE(int idEntreprise, String nom, String secteur,
                      String formeJuridique, LocalDate dateCreation,
                      double capital, int idProprietaire) {
        this.idEntreprise = idEntreprise;
        this.nom = nom;
        this.secteur = secteur;
        this.formeJuridique = formeJuridique;
        this.dateCreation = dateCreation;
        this.capital = capital;
        this.idProprietaire = idProprietaire;
    }

    // Getters & Setters

    public int getIdEntreprise() {
        return idEntreprise;
    }

    public void setIdEntreprise(int idEntreprise) {
        this.idEntreprise = idEntreprise;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getSecteur() {
        return secteur;
    }

    public void setSecteur(String secteur) {
        this.secteur = secteur;
    }

    public String getFormeJuridique() {
        return formeJuridique;
    }

    public void setFormeJuridique(String formeJuridique) {
        this.formeJuridique = formeJuridique;
    }

    public LocalDate getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(LocalDate dateCreation) {
        this.dateCreation = dateCreation;
    }

    public double getCapital() {
        return capital;
    }

    public void setCapital(double capital) {
        this.capital = capital;
    }

    public int getIdProprietaire() {
        return idProprietaire;
    }

    public void setIdProprietaire(int idProprietaire) {
        this.idProprietaire = idProprietaire;
    }

    @Override
    public String toString() {
        return "Entreprise{" +
                "idEntreprise=" + idEntreprise +
                ", nom='" + nom + '\'' +
                ", secteur='" + secteur + '\'' +
                ", formeJuridique='" + formeJuridique + '\'' +
                ", dateCreation=" + dateCreation +
                ", capital=" + capital +
                ", idProprietaire=" + idProprietaire +
                '}';
    }
}

File: src/Entity/Entreprise.php
<?php

namespace App\Entity;

use App\Repository\EntrepriseRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: EntrepriseRepository::class)]
#[ORM\Table(name: 'entreprises')]
class Entreprise
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_entreprise')]
    private ?int $id = null;

    #[ORM\Column(length: 150)]
    private ?string $nom = null;

    #[ORM\Column(length: 100, nullable: true)]
    private ?string $secteur = null;

    #[ORM\Column(name: 'forme_juridique', length: 50, nullable: true)]
    private ?string $formeJuridique = null;

    #[ORM\Column(name: 'date_creation', type: Types::DATE_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $dateCreation = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 15, scale: 2, nullable: true)]
    private ?string $capital = null;

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'entreprises')]
    #[ORM\JoinColumn(name: 'id_proprietaire', referencedColumnName: 'id_utilisateur', nullable: false)]
    #[Assert\NotNull(message: "Le propriétaire est obligatoire.")]
    private ?User $proprietaire = null;

    public function getProprietaire(): ?User { return $this->proprietaire; }
    public function setProprietaire(?User $user): static { $this->proprietaire = $user; return $this; }

    #[ORM\Column(nullable: true)]
    #[Assert\Range(
        min: -90,
        max: 90,
        notInRangeMessage: "La latitude doit être comprise entre {{ min }} et {{ max }}."
    )]
    private ?float $latitude = null;

    #[ORM\Column(nullable: true)]
    #[Assert\Range(
        min: -180,
        max: 180,
        notInRangeMessage: "La longitude doit être comprise entre {{ min }} et {{ max }}."
    )]
    private ?float $longitude = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $adresse = null;

    #[ORM\OneToMany(targetEntity: Document::class, mappedBy: 'entreprise')]
    private Collection $documents;

    public function __construct()
    {
        $this->documents = new ArrayCollection();
    }

    public function __toString(): string
    {
        return (string) $this->nom;
    }

    /**
     * @return Collection<int, Document>
     */
    public function getDocuments(): Collection
    {
        return $this->documents;
    }

    public function addDocument(Document $document): static
    {
        if (!$this->documents->contains($document)) {
            $this->documents->add($document);
            $document->setEntreprise($this);
        }

        return $this;
    }

    public function removeDocument(Document $document): static
    {
        if ($this->documents->removeElement($document)) {
            // set the owning side to null (unless already changed)
            if ($document->getEntreprise() === $this) {
                $document->setEntreprise(null);
            }
        }

        return $this;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;
        return $this;
    }

    public function getSecteur(): ?string
    {
        return $this->secteur;
    }

    public function setSecteur(?string $secteur): static
    {
        $this->secteur = $secteur;
        return $this;
    }

    public function getFormeJuridique(): ?string
    {
        return $this->formeJuridique;
    }

    public function setFormeJuridique(?string $formeJuridique): static
    {
        $this->formeJuridique = $formeJuridique;
        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(?\DateTimeInterface $dateCreation): static
    {
        if ($dateCreation instanceof \DateTimeImmutable) {
            $this->dateCreation = \DateTime::createFromImmutable($dateCreation);
        } else {
            $this->dateCreation = $dateCreation;
        }
        return $this;
    }

    public function getCapital(): ?string
    {
        return $this->capital;
    }

    public function setCapital(?string $capital): static
    {
        $this->capital = $capital;
        return $this;
    }


    public function getLatitude(): ?float
    {
        return $this->latitude;
    }

    public function setLatitude(?float $latitude): static
    {
        $this->latitude = $latitude;
        return $this;
    }

    public function getLongitude(): ?float
    {
        return $this->longitude;
    }

    public function setLongitude(?float $longitude): static
    {
        $this->longitude = $longitude;
        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(?string $adresse): static
    {
        $this->adresse = $adresse;
        return $this;
    }
}
